pkg/apis/networkpolicy/v1beta1/types.go

```diff
@@ -104,6 +104,13 @@ type NetworkPolicy struct {
 	Rules []NetworkPolicyRule `json:"rules,omitempty" protobuf:"bytes,2,rep,name=rules"`
 	// AppliedToGroups is a list of names of AppliedToGroups to which this policy applies.
 	AppliedToGroups []string `json:"appliedToGroups,omitempty" protobuf:"bytes,3,rep,name=appliedToGroups"`
+	// List of directions that the NetworkPolicy relates to.
+	// Valid options are "In", "Out", or "In,Out".
+	// It corresponds to PolicyTypes of K8s NetworkPolicy.
+	// If this field is not specified, it will default based on the existence of Ingress or Egress rules.
+	// If a direction is present, but there's no rule in that direction, the selected Pods are isolated
+	// in that direction (default-deny).
+	Directions []Direction `json:"directions,omitempty" protobuf:"bytes,4,rep,name=directions"`
```

Comment:
Talked with @abhiraut. Now thinking should we just generate a rule to the our internal messages for the default-deny? Then we need not the direction fields (which looks weird to me actually).

Reply:
I'm considering this too, since we already convert rules in controller, doing all convert (including the special case for allow-all) in the same place should be more clear, will update the implementation.

Reply:
Updated, PTAL.